dash/quorum/validator_conn_executor.go

```diff
@@ -88,15 +97,40 @@ func NewValidatorConnExecutor(
 	baseService := service.NewBaseService(logger, validatorConnExecutorName, vc)
 	vc.BaseService = *baseService
 
-	return vc
+	for _, opt := range opts {
+		err := opt(vc)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return vc, nil
+}
+
+// WithValidatorsSet sets the validators and quorum-hash as default values
+func WithValidatorsSet(valSet *types.ValidatorSet) func(vc *ValidatorConnExecutor) error {
+	return func(vc *ValidatorConnExecutor) error {
+		if len(valSet.Validators) == 0 {
+			return errValidatorsForUpdateInvalid
```

Comment:
Are you sure this is a breaking condition?

IMO ValidatorConnExecutor should simply "do nothing" (except logging) if it gets empty validator set, and wait for a correct validator update.